internal/plugins/helm/v1/init.go

```diff
@@ -109,102 +117,89 @@ Writes the following files:
+	p.config = c
 
-// Run will call the plugin actions
-func (p *initSubcommand) Run() error {
-	// Set values in the config
-	if err := p.config.SetProjectName(p.projectName); err != nil {
-		return err
-	}
 	if err := p.config.SetDomain(p.domain); err != nil {
 		return err
 	}
 
-	// Check if the project name is a valid k8s namespace (DNS 1123 label).
-	if p.config.GetProjectName() == "" {
+	// Assign a default project name
+	if p.projectName == "" {
 		dir, err := os.Getwd()
 		if err != nil {
 			return fmt.Errorf("error getting current directory: %v", err)
 		}
-		if err = p.config.SetProjectName(strings.ToLower(filepath.Base(dir))); err != nil {
-			return err
-		}
-	}
-
-	if err := cmdutil.Run(p); err != nil {
-		return err
-	}
-
-	// Run SDK phase 2 plugins.
-	if err := p.runPhase2(); err != nil {
-		return err
-	}
-
-	// If API creation is configured, run the 'create api' subcommand.
-	if p.apiSubc.options != emptyCreateOptions {
-		p.apiSubc.options.Domain = p.domain
-		if err := p.apiSubc.Run(); err != nil {
-			return err
-		}
+		p.projectName = strings.ToLower(filepath.Base(dir))
 	}
-
-	return nil
-}
-
-// SDK phase 2 plugins.
-func (p *initSubcommand) runPhase2() error {
-	if err := manifestsv2.RunInit(p.config); err != nil {
-		return err
+	// Check if the project name is a valid k8s namespace (DNS 1123 label).
+	if err := validation.IsDNS1123Label(p.projectName); err != nil {
+		return fmt.Errorf("project name (%s) is invalid: %v", p.projectName, err)
 	}
-	if err := scorecardv2.RunInit(p.config); err != nil {
+	if err := p.config.SetProjectName(p.projectName); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// Validate perform the required validations for this plugin
-func (p *initSubcommand) Validate() error {
-	if err := validation.IsDNS1123Label(p.config.GetProjectName()); err != nil {
-		return fmt.Errorf("project name (%s) is invalid: %v", p.config.GetProjectName(), err)
-	}
-
-	return nil
-}
-
-// GetScaffolder returns cmdutil.Scaffolder which will be executed due the RunOptions interface implementation
-func (p *initSubcommand) GetScaffolder() (cmdutil.Scaffolder, error) {
-	return scaffolds.NewInitScaffolder(p.config), nil
+func (p *initSubcommand) Scaffold(fs afero.Fs) error {
+	scaffolder := scaffolds.NewInitScaffolder(p.config)
+	scaffolder.InjectFS(fs)
+	return scaffolder.Scaffold()
 }
 
 // PostScaffold will run the required actions after the default plugin scaffold
 func (p *initSubcommand) PostScaffold() error {
-	if p.apiSubc.options == emptyCreateOptions {
+	doAPI := p.options.Group != defaultGroupFlagValue ||
+		p.options.Version != defaultVersionFlagValue ||
+		p.options.Kind != defaultKindFlagValue ||
+		p.apiSubcommand.options.chartOptions.Chart != defaultHelmChart
+	if !doAPI {
 		fmt.Printf("Next: define a resource with:\n$ %s create api\n", p.commandName)
+	} else {
+		args := []string{"create", "api"}
+		if p.options.Group != defaultGroupFlagValue {
```

Comment:
If we define that the default group should be an example then it will not work. 
So, IMO the if need to be != "" and also we need to trim the spaces

Reply:
See https://github.com/operator-framework/operator-sdk/pull/4581#discussion_r588455340